[AutoDiff] Mark SILGen derivative function thunks as "always inline".

dan-zheng opened the pull request:
SILGen derivative function thunks must be publicly exposed and cannot be
`[transparent]`. Instead, mark them as "always inline" for optimization.

Resolves [TF-925](https://bugs.swift.org/browse/TF-925).

---

Depends on https://github.com/apple/swift/pull/27752; rebase after it is merged.

rxwei commented on include/swift/SIL/SILFunctionBuilder.h:
```diff
@@ -79,7 +79,9 @@ class SILFunctionBuilder {
       IsThunk_t isThunk = IsNotThunk,
-      SubclassScope subclassScope = SubclassScope::NotApplicable);
+      SubclassScope subclassScope = SubclassScope::NotApplicable,
+      // SWIFT_ENABLE_TENSORFLOW
```
Why not just call ->setInlineStategy()?

dan-zheng replied:
Done in https://github.com/apple/swift/pull/27754/commits/0e7f03f5c79a5a3d17a227a52e33e5c803c880df.

dan-zheng committed:
[AutoDiff] Mark SILGen derivative function thunks as "always inline".

SILGen derivative function thunks must be publicly exposed and cannot be
`[transparent]`. Instead, mark them as "always inline" for optimization.

dan-zheng commented:
@swift-ci Please test tensorflow

dan-zheng commented:
@swift-ci Please test tensorflow

dan-zheng commented:
@swift-ci Please test tensorflow
